tools/file_replace.zig

```diff
@@ -0,0 +1,62 @@
+// NOTE(blackedout): Modified from https://ziglang.org/learn/build-system/#project-tools (2025-10-29)
+
+const std = @import("std");
+
+const usage =
+	\\Usage: ./file_replace pattern replacement file
+;
+
+pub fn main() !void {
+	var arena_state = std.heap.ArenaAllocator.init(std.heap.page_allocator);
+	defer arena_state.deinit();
+	const arena = arena_state.allocator();
+
+	const args = try std.process.argsAlloc(arena);
+	if(args.len != 4) {
+		try std.fs.File.stdout().writeAll(usage);
+		return std.process.cleanExit();
+	}
+
+	const pattern = args[1];
+	if(pattern.len == 0) {
+		fatal("pattern is empty (not allowed)", .{});
+	}
+	const replacement = args[2];
+	const filePath = args[3];
+
+	const outputFilePath = std.mem.concat(arena, u8, &.{filePath, ".tmp"}) catch |err| {
+		fatal("path allocation failed: {s}", .{@errorName(err)});
+	};
```

Comment:
Also for all allocations just use try. It is quite unlickely to happens and if it does, then the zig stanrdard library will print the error name anyways, that should be good enough for cases like this.